ipa-core/src/protocol/ipa_prf/shuffle/mod.rs

```diff
@@ -34,6 +34,16 @@ pub mod malicious;
 mod sharded;
 pub(crate) mod step;
 
+/// This struct stores some intermediate messages during the shuffle.
+/// In a maliciously secure shuffle,
+/// these messages need to be checked for consistency across helpers.
+/// `H1` stores `x1`, `H2` stores `x2` and `H3` stores `y1` and `y2`.
+#[derive(Debug, Clone)]
+struct IntermediateShuffleMessages<S> {
```

Comment:
any thoughts on making it an enum instead? seems that it can prevent some dumb errors where H1 can actually assign both